elife_api_validator/media_type.py

```diff
@@ -0,0 +1,88 @@
+import re
+
+from elife_api_validator.exceptions import MissingContentType
+
+
+class MediaType(object):
+    params = {}
+    type = ''
+
+    def __init__(self, content_type: str) -> None:
+        """
+        >>> mt = MediaType(content_type='application/vnd.elife.article-list+json; version=1')
+
+        :param content_type: str
+        """
+
+        self._content_type = content_type
+        if self._content_type:
+            self.parse_content_type(self._content_type)
+        else:
+            raise MissingContentType('Please supply a content type')
+
+    @property
+    def content_type(self):
+        return self._content_type
+
+    def get_type_str(self) -> str:
+        """Returns a formatted content type string.
+
+        >>> media_type = MediaType('application/vnd.elife.article-list+json; version=1')
+        >>> print(media_type.get_type_str())
+        'article-list.v1'
+
+        :rtype: str
+        """
+        # CHANGE TO REGEX SEARCH !!
+        type_str = self.type.split('.')[-1].replace('application/', '').replace('+json', '')
```

Comment:
yes, unclear at the moment what this is doing

Reply:
Now being found via regex:
` result = re.search(r'(?<=\.)[a-z-]*?(?=\+)', media_type.type)`